sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -238,6 +239,19 @@ public DefaultAzureCredentialBuilder disableInstanceDiscovery() {
         return this;
     }
 
+    /**
+     * Specifies environment variables that must be present when building the credential.
+     * If any of the specified environment variables are missing, {@link #build()} will throw an 
+     * {@link IllegalStateException}.
+     *
+     * @param envVars the names of environment variables that must be present
+     * @return An updated instance of this builder with the required environment variables set as specified.
+     */
+    public DefaultAzureCredentialBuilder requireEnvVars(String... envVars) {
```

Comment:
Should we consider using an ExpandableStringEnum? Users of this API will not know what env vars are available, make spelling/casing errors that can result in unexpected errors. Having a known list of values would be easier to use and also gives the flexibility of adding any string.

Reply:
We already have the well-known Identity env vars defined in Configuration class. 
https://github.com/Azure/azure-sdk-for-java/blob/348c25ff9a7230d3ec04f9378345c7b4db7d3c43/sdk/core/azure-core/src/main/java/com/azure/core/util/Configuration.java#L103
We need to add one to it, that will be a follow up PR for Core's October release.

Reply:
Added Expandable String Enum